### PR TITLE
PLAT-119495: Can't close a Drawer

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added SpotlightContainerDecorator to Drawer component, updated props and ui tests.

### Links
[//]: # (Related issues, references)
PLAT-119495

### Comments

### CODE REVIEW

**Comment by daniel-stoian-lgp** [comment]
I appreciate the effort to add the missing documentation. However, there is a PR open for this and it includes the missing drawer documenttion. https://github.com/enactjs/agate/pull/499/files
Can you use the docs  for Drawer from that PR? I am asking this because those js docs were already approved by a reviewer.

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by 0x64 — APPROVED** [review]

**Comment by 0x64** [comment]
@stanca-pop-lgp, Could you please add `onClose` handler for `Drawer`s in qa-a11y sample to close a drawer by pressing ESC key or by clicking out of a drawer area?

For example, change this

> \<Drawer open={open[0]}>

to

> \<Drawer open={open[0]} onClose={handleOpen(0, false)}>

File: Drawer/Drawer.js [hunk 1]
```diff
@@ -80,6 +92,14 @@ const DrawerBase = kind({
 		 */
 		onHide: PropTypes.func,
 
+		/**
+         * Called after the drawer's "show" transition finishes.
```
Comment (jeonghee27) [inline Drawer/Drawer.js:96]: Please check indentation.

File: Drawer/Drawer.js [hunk 2]
```diff
@@ -96,7 +116,31 @@ const DrawerBase = kind({
...
+         * * `'none'` - Spotlight can move freely within and beyond the drawer
+         * * `'self-first'` - Spotlight should prefer components within the drawer over
+         *   components beyond the drawer, or
+         * * `'self-only'` - Spotlight can only be set within the drawer
+         *
+         * @type {('none'|'self-first'|'self-only')}
+         * @default 'self-first'
+         * @public
+         */
+		spotlightRestrict: PropTypes.oneOf(['none', 'self-first', 'self-only'])
```
Comment (jeonghee27) [inline Drawer/Drawer.js:143]: there is no defaultProps for spotlightRestrict

**Comment by jeonghee27** [comment]
:shipit:

**Pull request merged**